Week-2/assignment2.2/src/index.html

```diff
@@ -0,0 +1,108 @@
+<!DOCTYPE html>
+<html lang="en">
+
+<head>
+  <meta charset="UTF-8">
+  <meta http-equiv="X-UA-Compatible" content="IE=edge">
+  <meta name="viewport" content="width=device-width, initial-scale=1.0">
+  <title>Assignment 2.2</title>
+  <link rel="icon" type="image/x-icon" href="./resources/favicon-2.2.png">
+  <style>
+    html {
+        width: 100%;
+        height: 100%;
+        margin: 0;
+        padding: 0;
+    }
+
+    body {
+        display: flex;
+        flex-direction: column;
+        width: 100%;
+        height: 100%;
+        margin: 0;
+        padding: 0;
+        align-items: center;
+    background-color: #f5f5f5;
+    }
+
+    header {
+height: 10rem;
+padding: 2rem;
+margin: 2rem;
+    }
+
+main {
+    position: relative;
+    display: flex;
+    justify-self: center;
+    align-items: center;
+    justify-content: center;
+}
+
+img {
+    width: 10rem;
+}
+
+#container {
+    position: absolute;
+    width: 15rem;
+    height: 15rem;
+    display: flex;
+    flex-wrap: wrap;
+}
+
+#container > div {
+    flex: 50%;
+    transition-property: transform;
+    transition-duration: 0.5s;
+}
+
+#d00 {
+    background-color: #F83D3A;
+}
+
+#d01 {
+    background-color: #FFEB3B;
+}
+
+#d10 {
+    background-color: #3333FF;}
+
+#d11 {
+    background-color: #00B800;
+}
+
+#container:hover #d00 {
+    transform: translateX(-7.5rem) rotate(90deg);
+}
+
+#container:hover #d10 {
+    transform: translateX(-7.5rem) rotate(-90deg);
+}
+
+#container:hover #d01 {
+    transform: translateX(7.5rem) rotate(-90deg);
+}
+
+#container:hover #d11 {
+    transform: translateX(7.5rem) rotate(90deg);
```

Comment:
this isnt the animation which was in the question. the blocks revolved around one corner (made a 360 degree rotation) and translated away from the center with the distance being their width. Please check.

Reply:
yes, I should have used `transform-origin` instead of `translate`. I have corrected it! pls have a look. Thanks.